docs/sphinx/heps/hep2.rst

```diff
@@ -0,0 +1,312 @@
+  - ``hpx::execution::then_execute``?
+  - ``hpx::execution::bulk_sync_execute``?
+  - ``hpx::execution::bulk_async_execute``?
+  - ``hpx::execution::bulk_then_execute``?
+  - ``hpx::execution::sequenced_executor``
+  - ``hpx::execution::parallel_executor``
+  - all the other executors? or just a selected subset?
+  - ``hpx::execution::service_executor``
+  - ``hpx::experimental::execution::thread_pool_executor``?
+  - ``hpx::execution::current_executor`` (not sure we need this)
+  - ``hpx::execution::is_executor`` and friends
+  - ``hpx::execution::static_chunk_size`` and other executor parameters
+
+- ``hpx/mutex.hpp``
+  - ``hpx::mutex``
+  - ``hpx::no_mutex``
+  - ``hpx::recursive_mutex``
+  - ``hpx::shared_mutex``
+  - ``hpx::spinlock``
+  - ``hpx::call_once``
+
+- ``hpx/condition_variable.hpp``
+  - ``hpx::condition_variable``
+
+- ``hpx/semaphore.hpp``
+  - ``hpx::counting_semaphore``
+  - ``hpx::sliding_semaphore``
+
+- ``hpx/barrier.hpp``
+  - ``hpx::barrier``
+
+- ``hpx/latch.hpp``
+  - ``hpx::latch``
+
+- ``hpx/channel.hpp``
+  - ``hpx::channel``
+
+- ``hpx/spmd_block.hpp``
+  - ``hpx::spmd_block``
+
+- ``hpx/task_block.hpp``
+  - ``hpx::task_block``
+
+Distributed
+...........
+
+- ``hpx/distributed/future.hpp``
+  - distributed equivalents of ``hpx::promise``, ``hpx::async``, etc.
+
+- ``hpx/distributed/algorithm.hpp``
+  - distributed equivalents of parallel algorithms
+
+- ``hpx/distributed/algorithm/for_loop.hpp``
+  - distributed equivalents of parallel algorithms
+
+- ``hpx/distributed/latch.hpp``
+  - ``hpx::distributed::latch``
+
+- ``hpx/distributed/barrier.hpp``
+  - ``hpx::distributed::barrier``
+
+- ``hpx/distributed/channel.hpp``
+  - ``hpx::distributed::channel``
+
+- etc. to match local headers and namespaces
+
+- ``hpx/distributed/components.hpp``
+  - ``HPX_REGISTER_COMPONENT``
+  - any other classes, functions, and macros that might be required to write components?
+
+- ``hpx/distributed/actions.hpp``
+  - ``HPX_REGISTER_ACTION``
+  - any other classes, functions, and macros that might be required to write actions?
+
+- ``hpx/distributed/x.hpp`` each in a separate header like channel, latch, etc.?
+  - ``hpx::distributed::all_to_all``
+  - ``hpx::distributed::all_reduce``
+  - ``hpx::distributed::broadcast``
+  - ``hpx::distributed::fold``
+  - ``hpx::distributed::gather``
+  - ``hpx::distributed::reduce``
+  - ``hpx::distributed::spmd_block``
+
+- ``hpx/distributed/checkpoint.hpp``?
+  - ``hpx::distributed::checkpoint`` stable?
+  - this is not necessarily distributed only...
+
+- ``hpx/experimental/distributed/resiliency.hpp``?
+  - ``hpx::experimental::distributed::async_replay`` and other resiliency?
+  - ready for public use or still experimental?
+  - the standard libraries use ``std::experimental::x``, perhaps we should just
+    follow that instead? the argument for ``hpx::x::experimental`` wasn't that
+    strong anyway...
+  - this is not necessarily distributed only...
```

Comment:
This is not distributed at all. All functionality is purely built on local facilities (no actions).

Reply:
Then the question is how stable this and checkpoint are. `hpx::experimental` or `hpx`? I.e. is there at least one active user for either of them, and is the interface likely to still need significant changes?

Reply:
Let's leave it in experimental. The new executor APIs will allow for this in a much more integrated way.